### DISCUSSION TITLE
Navigation

### DISCUSSION THREAD

**Discussion by Sonwon1** [osmandapp/OsmAnd#17942] [Q&A]
I am trying to enter the Google map navigation route into OsmAnd Navigation and it is not working well. First it is difficult to find a location. Entering in Brunau, Austria does not bring back Brunau, Austria. Google Maps has Brunau, Austria? I am giving up at my third entry.

Please give this a try and tell me how to make this route without a lot of pain.

Android 13, Samsung Galaxy Tab Active 3, version 4.5.10.

**Comment by Sonwon1** [comment]
Here is the route I am trying to enter into OsmAnd.

https://www.google.com/maps/dir/Hotel+Krone+Tirol,+6+W%C3%A4ngler+Str.,+Lechaschau+6600,+Austria/Reutte,+Austria/Brunau,+Tyrol,+Austria/TOP+Mountain+Motorcycle+Museum+Crosspoint,+Timmelsjochstrasse,+Hochgurgl,+Austria/Top+Mountain+Crosspoint+Restaurant,+Timmelsjochstra%C3%9Fe,+Hochgurgl,+Austria/Gasthof+Schlossberg,+JaufenStra%C3%9Fe,+48,+39015+St.+Leonhard+in+Passeier,+Autonome+Provinz+Bozen+-+S%C3%BCdtirol,+Italy/@47.096757,10.7465091,9.75z/data=!3m1!5s0x479cf13a69cd4c1b:0x246fa58e014fbaea!4m40!4m39!1m5!1m1!1s0x479cf13a68dfc205:0xbc8b5c50534ed15e!2m2!1d10.7057136!2d47.4893241!1m5!1m1!1s0x479cfa7e6a446463:0xb386b892d0c6daa5!2m2!1d10.71865!2d47.48879!1m5!1m1!1s0x479cdf858af4a5f5:0xcc64ff5fad1ac3b1!2m2!1d10.85995!2d47.22098!1m5!1m1!1s0x4782ccaa2b89dd41:0x3aca6e49a86d3a54!2m2!1d11.0528693!2d46.9096504!1m5!1m1!1s0x4782cdece83b3019:0xcc1a062e52657f11!2m2!1d11.0531852!2d46.9093554!1m5!1m1!1s0x4782b3de3d8dcc8b:0x176f904b5e05a4fc!2m2!1d11.2582483!2d46.8198623!2m1!1b1!3e0?entry=ttu

**Comment by Sonwon1** [comment]
I loaded Brouter and now it routes the same as Google maps. However when I save the route and reload I have to use Attach to roads to fix all of the turns. Attach to roads is very slow, to slow to be useful in my opinion. It seems like when I create a Broute OsmAnd should be able to save that exact route without needing Attach to Roads when the route is reloaded.

When using OsmAnd for routing OsmAnd believes there is unpaved road between Hotel Krone Tirol, Wängler Str. 6, 6600 Lechaschau, Austria and Brunau, 6433, Austria (I figured out how to find Brunau in the search, the results are a bit confusing.). I checked with Google Maps and there is no unpaved road on the Google Route as mapped. Broute also does not see unpaved roads.

I tried adding points to force the OsmAnd route where I wanted it to go and now I am getting Unfortunately, OsmAnd could not determine a route for your settings. If I just leave OsmAnd alone it finally comes back with a route that is useless. There is a problem routing directly from Sölden to Gasthof Schlossberg. Google Maps routes through the pass.

We have a trip planned for next month and I was going to use OsmAnd. However with the navigation routing in the current state I will have to use the Garmin instead. I will hope these issues will be resolved in a future update. Please let me know what you would need from me to fix these issues?

**Reply by DmitryAlexei** [reply]
- Attach to roads is needed for more precise turn information and notifications. Also with this function you can see *[Route Details](https://docs.osmand.net/docs/user/navigation/setup/route-details)*.
- As data source *[Openstreetmap](https://www.openstreetmap.org/)* is used.
- Please note, that you can also use [Online routing engines](https://docs.osmand.net/docs/user/navigation/routing/online-routing) inside OsmAnd (internet connection is required).

**Reply by Sonwon1** [reply]
Thanks for the reply. I already knew all of this and it does not help to make navigation usable in my case.

The Attach to roads is to slow for long routes in my opinion. There should be a way to optimize the process to reduce processing time. If I could see the algorithm maybe I would see a faster method. Is that in a flow chart that could be shared?

The data source is not the problem since Google Maps (Google proprietary) and brouter (Openstreetmap) route down that road. I will pull that map section up later today and see what type of road Openstreetmap is showing to validate the road type.

I will be traveling in locations where mobile data will not always work so I prefer offline mode. This is why I ditched Google Maps because it can sometimes become erratic when it looses the data connection.

I believe the Brouter route should be savable with turn information. I need to check again however I believe a list of turns are generated and the list should be savable and then reload and rebuild the route from the turn list. Is there a technological reason why this would not work?

I want to help make this work because it it is unusable for me at the moment. And I really do not want to go back to the Garmin and Google Maps is unusable for the lack of an offline mode. OsmAnd is the best navigation software I have seen excluding the problems.

**Reply by Sonwon1** [reply]
Well this took two hours to duplicate. There appears to be some type of interaction between Unpaved roads and Consider temporary limitations. With Consider temporary limitations off it always routes around that road section. The road I want to follow is the google maps link I posted above. And like I said Brouter routes the same as Google.

1. Onscreen Navigation button set the following route parameters:
2. Avoid Roads - all off
3. Fuel-efficiant way - on
4. Consider temporary limitations - on
5. Cancel navigation
6. Close OsmAnd

7. Start OsmAnd
8. Onscreen Navigation button
9. From - Reutte, Austria
10. Gasthof Schlossberg, JaufenStraße, 48, 39015 St. Leonhard in Passeier, Autonome Provinz Bozen - Südtirol, Italy
11. Notice the first 25% of the route
12. Now goto to Settings in the Navigation menu
13. Select Avoid roads...
14. Turn on No unpaved roads and Apply
15. If you get check the necessary maps tap on that
16. Notice the top route has complete changed

I scanned that whole route and did not see any unpaved roads! Should I report this as an Issue?

**Reply by Sonwon1** [reply]
When I use Brouter for navigation it generates a turn by turn list so this should be usable to rebuild a restored Broute route.

**Reply by Sonwon1** [reply]
Attach to the roads is still painfully slow even with Safe mode disabled.
